Docker compose and dockerfile updates

orangewolf opened the pull request:
get a building dockerfile and a docker-compose to make building and pushing easy

orangewolf committed:
get a building dockerfile and a docker-compose to make building and pushing easy

orangewolf committed:
back in actions

orangewolf committed:
build the container only

orangewolf committed:
add target

orangewolf committed:
working on helm deploy

orangewolf committed:
helm based deployment

orangewolf committed:
working deployment

alishaevn commented on .github/workflows/deploy.yml:
```diff
@@ -1,25 +1,26 @@
-# we are running the deploy through vercel now so we don't need this github action
-# TODO(alishaevn): fix any potential sentry error's if we switch from vercel to an in house deploy infrastructure
+we are running the deploy through vercel now so we don't need this github action
+TODO(alishaevn): fix any potential sentry error's if we switch from vercel to an in house deploy infrastructure
 
```
```suggestion
```

alishaevn replied:
assuming you meant to delete this entirely.

alishaevn commented on .github/workflows/build.yml:
```diff
@@ -1,78 +1,24 @@
-#             ${{ env.REGISTRY }}/${{ env.REPO_LOWER }}:${{ env.TAG }}
+jobs:
+  build:
+    uses: scientist-softserv/actions/.github/workflows/build.yaml@v0.0.14
```
we're using `v0.0.15` in the deploy file. is this intentional?

orangewolf replied:
no, it can be 0.0.15. they do the same thing for this part

alishaevn replied:
```suggestion
    uses: scientist-softserv/actions/.github/workflows/build.yaml@v0.0.15
```

alishaevn commented on .gitignore:
```diff
@@ -7,9 +7,13 @@ node_modules/
 # Where ENV secrets are stored. refer to the "Note" at the link below for details on the use of this file
 # https://nextjs.org/docs/basic-features/environment-variables#default-environment-variables
 .env.local
+.env.development
```
why'd you add this file? what's in it?

orangewolf replied:
.env.development is our standard env file for all apps. it means we can use stack_car on it. .env.local is similar in purpose, but just trying to keep things consistant

alishaevn commented on ops/production-deploy.tmpl.yaml:
```diff
@@ -0,0 +1,36 @@
+nextAuthUrl: "https://phenovista.softserv.cloud/api/auth"
```
is this temporary? (along with the other references to it below) they want the site to point at store.phenovista.com.

ref: [this ticket](https://github.com/scientist-softserv/phenovista-digital-storefront/issues/17)

orangewolf replied:
yes, I dont have access to that url so we need something to get the deploy all figured out

orangewolf committed:
Update .github/workflows/deploy.yml

Co-authored-by: Alisha Evans <alishaevn2@gmail.com>

alishaevn committed:
use `v0.0.15` for the build actions

being updated for consistency. this is the same version being used in the deploy file.

alishaevn commented on .github/workflows/build.yml:
```diff
@@ -1,78 +1,24 @@
-#             ${{ env.REGISTRY }}/${{ env.REPO_LOWER }}:${{ env.TAG }}
+jobs:
+  build:
+    uses: scientist-softserv/actions/.github/workflows/build.yaml@v0.0.15
```
```suggestion
    uses: scientist-softserv/actions/.github/workflows/build.yaml@v0.0.14
```

alishaevn committed:
reverting to v0.0.14 in the build workflow

when I changed this version from 14 to 15, the build action failed with the messaging below:

"Failed to load /home/runner/work/phenovista-digital-storefront/phenovista-digital-storefront/.env.development: open /home/runner/work/phenovista-digital-storefront/phenovista-digital-storefront/.env.development: no such file or directory"
ref: https://github.com/scientist-softserv/phenovista-digital-storefront/actions/runs/6892743232/job/18750807293#step:32:28

the .env.development is gitignored so I'm unsure why this error didn't occur while using 14 the first time around. going to test it out though.

alishaevn commented:
ref:
- https://github.com/scientist-softserv/webstore/issues/322
